open_parser/base.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+import requests
+
+CAMBIO_UPLOAD_URL = (
+    "https://p1iz3c1c77.execute-api.us-west-2.amazonaws.com/v1/cambio_api/upload"
+)
+CAMBIO_EXTRACT_URL = (
+    "https://p1iz3c1c77.execute-api.us-west-2.amazonaws.com/v1/cambio_api/extract"
+)
+
+
+class OpenParser:
+    def __init__(self, apiKey="") -> None:
+        self._uploadurl = CAMBIO_UPLOAD_URL
+        self._extracturl = CAMBIO_EXTRACT_URL
+        self._request_header = {"x-api-key": apiKey}
+
+    def setAPIKey(self, apiKey):
+        self._request_header = {"x-api-key": apiKey}
+
+    def extract_pdf_content(self, file_path):
+        user_id, job_id, s3_key = self._request_and_upload_by_apiKey(file_path)
+        result = self._request_file_extraction(user_id, job_id, s3_key)
+        return result["file_content"]
+
+    def _error_handler(self, response):
+        if response.status_code == 403:
+            raise Exception("Invalid API Key")
+        elif response.status_code == 429:
+            return Exception("API Key limit exceeded")
+        else:
+            return Exception(f"Error: {response.status_code} {response.text}")
+
+    def _request_and_upload_by_apiKey(self, file_path):
```

Comment:
qq: do we need a job_type to tell the backend regarding what kind of job is this especially this is for "document_extraction" to use "document_extraction_lambda".